runtime/io-error.cpp

```diff
@@ -76,6 +77,32 @@ bool IoErrorHandler::GetIoMsg(char *buffer, std::size_t bufferLength) {
     ToFortranDefaultCharacter(buffer, bufferLength, msg);
     return true;
   }
+
+  char *newBuf;
+  // Following code is taken from llvm/lib/Support/Errno.cpp
+  // in LLVM v9.0.1
+#if HAVE_STRERROR_R
+  // strerror_r is thread-safe.
+#if defined(__GLIBC__) && defined(_GNU_SOURCE)
+  // glibc defines its own incompatible version of strerror_r
+  // which may not use the buffer supplied.
+  newBuf = ::strerror_r(ioStat_, buffer, bufferLength);
+#else
   return ::strerror_r(ioStat_, buffer, bufferLength) == 0;
+#endif
+#elif HAVE_DECL_STRERROR_S // "Windows Secure API"
```

Comment:
Perhaps move this case first in case glibc ever defined strerror_s?

Reply:
I'd rather keep this code similar to the one in `llvm/lib/Support/Errno.cpp`